draft-ietf-ppm-dap.md

```diff
@@ -1499,6 +1518,45 @@ its query:
 * For fixed-size tasks, the batch selector is the batch ID assigned sent in the
   response.
 
+### Collect Message Security
+
+Collect sub-protocol messages must be confidential and mutually authenticated.
+
+HTTPS provides confidentiality and authenticates the leader to the collector.
+Additionally, the leader encrypts its aggregate share to a public key held by
+the collector.
+
+Collectors MUST authenticate their requests to leaders using a scheme that meets
+the requirements in {{request-authentication}}.
+
+[[OPEN ISSUE: collector public key is currently in the task parameters, but this
+will have to change #102]]
+
+The collector and helper never directly communicate with each other, but the
+helper does transmit an aggregate share to the collector through the leader, as
+detailed in {{collect-aggregate}}. The aggregate share must be confidential from
+everyone but the helper and the collector.
+
+Confidentiality is achieved by having the helper encrypt its aggregate share to
+a public key held by the collector.
+
+There is no authentication between the collector and the helper. This allows the
+leader to:
+
+* Send collect parameters to the helper that do not reflect the parameters
+  chosen by the collector
+* Discard the aggregate share computed by the helper and then fabricate
+  aggregate shares that combine into an arbitrary aggregate result
+
+These are both attacks on soundness, which we already assume to hold only if
+both aggregators are honest, which puts these malicious-leader attacks out of
+scope.
```

Comment:
Point to security considerations here.

Reply:
I put in a reference to {{sec-considerations}} and use the word "correctness", which is currently the term that Security Considerations uses. That's wrong, but it'll get cleaned up in #306.

It also occurs to me that maybe everything in these "{Upload, Aggregate, Collect} Message Security" sections that isn't a MUST or a MAY should be punted to security considerations, since it's discussion, context and justification rather than prescription.

Reply:
I think it's OK to point to security considerations as needed in the normative text, especially when it clarifies certain design choices to the reader. That said, mnimizing where possible is a good idea. In particular I'm not sure I would call these sections ".* Message Security".